src/test/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/accession/services/JpaBasicSpringDataRepositoryDatabaseServiceTest.java

```diff
@@ -134,4 +134,24 @@ public void multipleUpdates() throws AccessionDoesNotExistException,
         assertTrue(versions.contains(3));
     }
 
+    @Test
+    public void testFindByAccessionAndVersion() throws AccessionDoesNotExistException,
+            HashAlreadyExistsException {
+        service.insert(Arrays.asList(
+                new AccessionWrapper("a2", "h1", TestModel.of("something2"), 1),
+                new AccessionWrapper("a2", "h2", TestModel.of("something2b"), 2),
+                new AccessionWrapper("a2", "h3", TestModel.of("something2c"), 2)));
```

Comment:
is this correct? There are two "a2" version 2 accessions

Reply:
Yes, it is intentional so that database can support merge.